salt/patroni/files/etc/default/etcd.j2

```diff
@@ -1,12 +1,19 @@
 {%- set own_ip                   = salt['mine.get'](grains.id, pillar_etcd.etcd_address_mine_function)[grains.id][0] %}
 
+{%- set own_ip_is_ipv6 = ':' in own_ip %}
+{%- if own_ip_is_ipv6 %}
+  {%- set formatted_peer_ip = '[' ~ own_ip ~ ']' %}
```

Comment:
Salt has an `ipwrap` filter, I suggest to use that instead of the if/set/else/set/.